### DISCUSSION TITLE
[FontFamily] Custom FontFamily not work when publish nativeaot

### DISCUSSION THREAD

**Discussion by ChisterWu** [AvaloniaUI/Avalonia#14256] [Q&A]
When I use  fontcollection to add new custom fontfamily, It can work normal. But when I publish fot nativeaot, It can't take effect.What do I need to do for it?

**To Reproduce**

1. Add FontCollection 
2. Debug or release
3. Publish NativeAot 

- OS: [ Windows, Mac, Linux]
- Version [V11-Preview 6]

**Comment by maxkatz6** [comment]
It's impossible to tell where your problem is.
NativeAOT by itself won't affect your embedded resources.
But if, for example, you have your resource/font in the separated assembly, which isn't directly referenced in the code, it might be removed by the compiler.
Can you provide more information?

**Comment by ChisterWu** [comment]
![image](https://user-images.githubusercontent.com/28770378/233572508-16a674d3-c68f-4ef1-bae5-2a4a8d8fe20b.png)
![image](https://user-images.githubusercontent.com/28770378/233573091-d7ed830d-b085-474e-9607-bc8254cb6e07.png)
this is my avalonia project, I add a font assembly and use ConfigureFonts add my custom fontfamily,  all the font files are AvaloniaResource, I add it for my desktop app, if i debug and release run this app, it work ok. But when i publish fot NativeAOT, the font is not work.
![image](https://user-images.githubusercontent.com/28770378/233573855-cc284aab-1f0e-4752-8304-a930a411692b.png)
I add the font assembly in the rd.xml.

![image](https://user-images.githubusercontent.com/28770378/233573974-feb778f5-92da-4d85-8f25-468d547f3fc8.png)
this is my app run with debug, the font is my custom font.

![image](https://user-images.githubusercontent.com/28770378/233574244-e6c5b64c-9f03-4058-8506-f04bd1d272c5.png)
this is my app which published for NativeAOT, font is not the custom font.

**Comment by ChisterWu** [comment]
You can get sourcecode from https://github.com/AvaloniaDeveloperOrg/AntDesign.Avalonia

**Comment by Gillibald** [comment]
Your default font family should be: `fonts:AntDesignFontFamilies#Alibaba PuHuiTi 2.0`

**Comment by ChisterWu** [comment]
![image](https://user-images.githubusercontent.com/28770378/233616804-7994638d-033f-4434-a97e-c61d59e4614f.png)
I add this fontfamily for windows style and usercontrol style .
![image](https://user-images.githubusercontent.com/28770378/233616972-aac48262-7dca-4007-a75a-6fb0e82a6de2.png)
And  I also add it for window,but not invalid。
![image](https://user-images.githubusercontent.com/28770378/233617256-6f2cb16e-309a-4b8c-b7a8-eb4e1bf4fd9e.png)
When I use this setting, it will crash

**Comment by Gillibald** [comment]
You could follow our approach of exposing the font collection via some AppBuilder extension that is defined in the asset assembly. This makes sure the assembly is referenced.

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Fonts.Inter/AppBuilderExtension.cs
